build.go

```diff
@@ -96,6 +96,14 @@ func Build(
 		}
 
 		logger.Process("Executing build process")
+
+		gemsLayer, err = gemsLayer.Reset()
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+		gemsLayer.Launch, gemsLayer.Build = entries.MergeLayerTypes("gems", context.Plan.Entries)
+		gemsLayer.Cache = gemsLayer.Build
```

Comment:
I don't think we want to bring this behavior back. It was an explicit decision previously that we shouldn't reset the entire layer when rebuilding as this can be very costly: https://github.com/paketo-buildpacks/bundle-install/pull/109.

Reply:
Ah yea, I couldn't find that issue when I was looking and thought I was going crazy. Thanks!